### PR TITLE
Implementing Bigtable Table.rename().

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

File: gcloud/bigtable/table.py [hunk 1]
```diff
@@ -139,6 +139,36 @@ def create(self, initial_split_keys=None):
...
+            zones, or projects.
+
+        .. note::
+
+            The Bigtable Table Admin API currently returns
+
+                ``BigtableTableService.RenameTable is not yet implemented``
+
+            when this method is used. It's unclear when this method will
+            actually be supported by the API.
```
Comment (tseaver) [inline gcloud/bigtable/table.py]: Doesn't that make this method an "attractive nuisance"?

**Comment by tseaver** [comment]
Implementation LGTM, given that we want to add the `rename` method at all.

File: gcloud/bigtable/table.py (hunk 1 above)
Comment (dhermes) [inline gcloud/bigtable/table.py]: Just trying to match the [service definition][1] (see [`RenameTableRequest`][2] def'n as well).

I actually haven't checked if the method works in awhile. Doing it now and will post back what I found out.

[1]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/5d84776473347513db0b6c44e4f8503837a0d9fa/bigtable-protos/src/main/proto/google/bigtable/admin/table/v1/bigtable_table_service.proto#L56-L58
[2]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/5d84776473347513db0b6c44e4f8503837a0d9fa/bigtable-protos/src/main/proto/google/bigtable/admin/table/v1/bigtable_table_service_messages.proto#L77-L84

Comment (dhermes) [inline gcloud/bigtable/table.py]:
```python
>>> try:
...     TABLE.rename('gcloud-pythoo-new-name')
... except Exception as exc:
...     pass
...
>>> exc
grpc.framework.interfaces.face.face.LocalError()
>>> exc.code
<StatusCode.UNIMPLEMENTED: 12>
>>> exc.details
'BigtableTableService.RenameTable is not yet implemented'
```

----

@nathanielmanistaatgoogle What is the difference between a `NetworkError` and `LocalError`?

Comment (nathanielmanistaatgoogle) [inline gcloud/bigtable/table.py]: Ostensibly `LocalError` indicates that you made a programming mistake on your side of the connection and `NetworkError` indicates that there was some fault in the connection, but the categorization is rather broad and it's the status code that should be considered most authoritative.

Comment (dhermes) [inline gcloud/bigtable/table.py]: So is that the intended behavior here? I've gotten a `NetworkError` for using 32 bytes in a field that was capped at 30 and gotten a `LocalError` for sending a fully valid request to a method that isn't implemented (though the proto doesn't mention this).

Comment (nathanielmanistaatgoogle) [inline gcloud/bigtable/table.py]: Maybe, but even if it were intended, it doesn't sound like it's being helpful. Do the descriptions of the status codes used in these two circumstances better align with the experience of the developer on the receiving end of them?

Comment (dhermes) [inline gcloud/bigtable/table.py]: For comparison, here is the output of the other [error case][1]

```python
>>> exc
NetworkError()
>>> exc.code
<StatusCode.INVALID_ARGUMENT: 3>
>>> exc.details
"Error in field 'cluster_id' : Invalid id for collection cluster : Length should be between [6,30], but found 32 (gcloud-python-cluster-0123456789)"
```

[1]: https://github.com/grpc/grpc/issues/2611#issuecomment-159721031

Comment (dhermes) [inline gcloud/bigtable/table.py]: @nathanielmanistaatgoogle The real concern is that we don't want the exceptions bubbling up to our users unless we know what to expect. Should I be catching everytime I run a method on a stub?

Comment (nathanielmanistaatgoogle) [inline gcloud/bigtable/table.py]: What the stub API specifies is that any RPC method call may raise `AbortionError` (or any of its subclasses). So if you don't want those propagating to callers of your system then yes, I believe you'll have to catch them.

Comment (dhermes) [inline gcloud/bigtable/table.py]: OK cool. At least there is a parent class we can grab on to if need be.

**Comment by dhermes** [comment]
@tseaver How to proceed?

**Comment by tseaver** [comment]
@dhermes Let's add a `.. note:` in the docstring that as of whichever API version we are supporting, the method is not implemented on the backend, and add an issue to review when we move to a new version.

**Comment by dhermes** [comment]
@tseaver Filed #1289 for followup. I added `v1` to the note about 

```
BigtableTableService.RenameTable is not yet implemented
```

LMK if we are a go to merge.

**Comment by tseaver** [comment]
LGTM

**Pull request merged**